axelrod/history.py

```diff
@@ -130,12 +134,13 @@ def flip_plays(self):
     def append(self, play, coplay):
         """Appends a new (play, coplay) pair an updates metadata for
         number of cooperations and defections, and the state distribution."""
+        # casts plays sent as integers into Action objects
+        play, coplay = Action(play), Action(coplay)
```

Comment:
I see that there are several castings to `Action`, but this is inefficient if rarely needed, and smells like something upstream is broken.

Reply:
hmm, you’re right that there may be a better solution - it’s more than just the transformer (there were also issues in history). they’re caused by Action objects being automatically cast by some functions due to being the IntEnum type. I might redesign so that the Action class is a regular Enum, but has an .int() method that converts it to an int when we want to do so on purpose; this gives the best of both worlds and also would work better with custom action sets (e.g. if someone wanted to create an Action class for the quantum prisoners dilemma, with C, D and Q actions)